### DISCUSSION TITLE
Add `DEFAULT SELECTION/SELECTIONS` option to  `Relation` and `Select` fields

### DISCUSSION THREAD

**Discussion by babakfp** [pocketbase/pocketbase#1759] [Q&A]
Hey
I couldn't find a way to do this. Is there something that I can do to set this up from the Admin UI?
Thanks

**Comment by ganigeorgiev** [comment]
Default values are not supported at the moment from the Admin UI. There were several issues and discussions in the past for this (https://github.com/pocketbase/pocketbase/issues/42, https://github.com/pocketbase/pocketbase/issues/1043, https://github.com/pocketbase/pocketbase/discussions/1622, etc.), but for now default values are left to the developers to specify on their own either client-side as part of the request data or server-side with [event hooks](https://pocketbase.io/docs/event-hooks/).

**Reply by babakfp** [reply]
### Default values (hasDefault)

I was thinking about this a bit, and I think the best thing to do is to give the user a choice to configure when they are going to use the default value and for what scenarios.

For example, fallback to default (when received `null|undefined`, when not received anything) and etc.

In [this comment](https://github.com/pocketbase/pocketbase/issues/42#issuecomment-1179666245) you said:

> it's not clear when the default will be applied:
> on create or create/update?

I don't know much about this kind of backend & database stuff, so did you mean should we use the default value when (creating a record) or when (creating and updating a record)? If so, I think the default could be (when creating and updating), and if the user wants to change it, they can do it...

> what about multipart/form-data requests? should users still send null as string (PocketBase supports both json and multipart/form-data content-types)?

Sorry, I don't know what this is😅.

> what about other zero values (0, [], "", etc.)?

As I said at the beginning, I think we can give the user the choice (the options) to configure this stuff. These options can be different for different field types...

### About even hooks

Is it possible to do this stuff with JavaScript? I don't know how to do this stuff...

---

Thanks

**Reply by babakfp** [reply]
Can we also have the feature to make some of the default select options required? So you can't later remove it from the select field items when editing the record...

**Reply by ganigeorgiev** [reply]
Sorry, as mentioned in the linked issues, defaults are not planned for now.

> Can we also have the feature to make some of the default select options required?

You can either use a before [model/record hook](https://pocketbase.io/docs/event-hooks/) or filter it as part of your API rule, (eg. `@request.data.someMultiSelectField ~ '"something"'`).
